arcs/src/algorithms/affine_transform.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{
+    primitives::{Point, Line,},
+    Vector,
+};
+use kurbo::Affine;
+
+pub trait AffineTransformable {
+    fn transform(&mut self, affine: Affine, base: Vector);
```

Comment:
Do we actually need the `base` parameter here? I would have thought if you need an operation to be relative to a particular point you'd incorporate that into your transform matrix.

Something else to keep in mind is this function will be the simplest building block for most geometric operations so it'll be really, really hot. If we push the `translate_to_base * affine * translate_back` dance up to the caller it means they can reuse the same `Affine` for multiple calls to `transform()` and there's less code to get in the way of inlining or auto-vectorisation.

Reply:
We don't *need* the `base` parameter, it's just *convenient*.

Removing it for better performance is something I can get behind, although I would *-at some point-* implement some convenience wrappers, as in my experience, composing low-level transforms can be quite error-prone for users.

But de-coupling this from the most basic implementation seems right to me.